File: programmers/Code/Printer.py
def solution(priorities, location):

    def isPossible(num):
        for idx,num2 in priorities:
            if num2>num: return False
        return True

    for i in range(len(priorities)):
        priorities[i]=(i,priorities[i])
    answer = 0
    while True:
        idx,num=priorities.pop(0)
        if isPossible(num):
            answer+=1
            if idx==location:
                return answer
        else:
            priorities.append((idx,num))
    return answer

print(solution([1,1,9,1,1,1],0))

File: programmers/Code/Printer.js
// function solution(priorities, location) {
//   let cnt=0;
//   let curr_idx=0;
//   while(true){
//     let target = priorities.shift();
//     if(priorities.filter(num => num>target).length !== 0){
//       priorities.push(target);
//     }
//     else{
//       cnt++;
//       if(curr_idx===location) return cnt;
//     }
//     curr_idx = curr_idx>=priorities.length ? curr_idx=0 : curr_idx + 1;
//   }
// }
// console.log(solution([2, 1, 3, 2],2));
// console.log(solution([1,1,9,1,1,1],0));